src/style/style.js

```diff
@@ -769,6 +769,17 @@ class Style extends Evented {
         return this._layers[id];
     }
 
+    /**
+     * Return a boolean specifying if the layer is valid (not deleted) with the given `id`.
+     *
+     * @param {string} id - id of the desired layer
+     * @returns {boolean} a boolean specifying if the given layer is valid
+     */
+    isValidLayer(id: string): boolean {
```

Comment:
nit: Can this function be renamed  to indicate that it is checking for the existence of a layer not its validity. Some ideas: `hasLayer` or `isCurrentLayer`